Gems/Atom/Tools/AtomToolsFramework/Code/Source/AssetSelection/AssetSelectionComboBox.cpp

```diff
@@ -78,34 +74,108 @@ namespace AtomToolsFramework
     {
         if (m_filterCallback && m_filterCallback(assetInfo))
         {
-            int index = findData(QVariant(QString(assetInfo.m_assetId.ToString<AZStd::string>().c_str())));
-            if (index >= 0 && index < count())
+            int index = findData(QVariant(QString(assetId.ToString<AZStd::string>().c_str())));
+            if (index >= 0)
             {
                 removeItem(index);
             }
         }
     }
+
+    void AssetSelectionComboBox::AddAsset(const AZ::Data::AssetInfo& assetInfo)
+    {
+        if (m_filterCallback && m_filterCallback(assetInfo))
+        {
+            addItem(
+                GetDisplayNameFromPath(AZ::RPI::AssetUtils::GetSourcePathByAssetId(assetInfo.m_assetId).c_str()),
+                QVariant(QString(assetInfo.m_assetId.ToString<AZStd::string>().c_str())));
+
+            RegisterThumbnail(assetInfo.m_assetId);
+        }
+    }
+
+    void AssetSelectionComboBox::RegisterThumbnail(const AZ::Data::AssetId& assetId)
+    {
+        if (m_thumbnailsEnabled)
+        {
+            AzToolsFramework::Thumbnailer::SharedThumbnailKey thumbnailKey =
+                MAKE_TKEY(AzToolsFramework::AssetBrowser::ProductThumbnailKey, assetId);
+            m_thumbnailKeys[assetId] = thumbnailKey;
+
+            connect(
+                thumbnailKey.data(), &AzToolsFramework::Thumbnailer::ThumbnailKey::ThumbnailUpdatedSignal, this,
+                [this, assetId]() { QueueUpdateThumbnail(assetId); });
+
+            QueueUpdateThumbnail(assetId);
+        }
+    }
+
+    void AssetSelectionComboBox::UpdateThumbnail(const AZ::Data::AssetId& assetId)
+    {
+        if (m_thumbnailsEnabled)
+        {
+            auto thumbnailKeyItr = m_thumbnailKeys.find(assetId);
+            if (thumbnailKeyItr != m_thumbnailKeys.end())
+            {
+                int index = findData(QVariant(QString(assetId.ToString<AZStd::string>().c_str())));
+                if (index >= 0)
+                {
+                    AzToolsFramework::Thumbnailer::SharedThumbnail thumbnail;
+                    AzToolsFramework::Thumbnailer::ThumbnailerRequestBus::BroadcastResult(
+                        thumbnail, &AzToolsFramework::Thumbnailer::ThumbnailerRequests::GetThumbnail, thumbnailKeyItr->second);
+                    if (thumbnail)
+                    {
+                        // Adding pixmaps for all of the states so they don't disappear when highlighting 
+                        const QPixmap pixmap = thumbnail->GetPixmap().scaled(iconSize(), Qt::KeepAspectRatio, Qt::SmoothTransformation);
+                        QIcon icon;
+                        icon.addPixmap(pixmap, QIcon::Normal);
+                        icon.addPixmap(pixmap, QIcon::Disabled);
+                        icon.addPixmap(pixmap, QIcon::Active);
+                        icon.addPixmap(pixmap, QIcon::Selected);
+                        setItemIcon(index, icon);
+                    }
+                }
```

Comment:
Can you verify how these look on displays with scaling > 100%?

Had to work on that some time ago to ensure the pixmaps are scaled correctly on highDpi screens, here's the PR: https://github.com/o3de/o3de/pull/4070

That added a helper function in [`AzQtComponents/Utilities/PixmapScaleUtilities.cpp`](https://github.com/o3de/o3de/pull/4070/files#diff-2751910d0e3a0e91e5098a4c2bdd5ccbc0dc369279227bf104f2d312719d3605) you can reuse if it helps:
```
QPixmap ScalePixmapForScreenDpi(
        QPixmap pixmap, QScreen* screen, QSize size, Qt::AspectRatioMode aspectRatioMode, Qt::TransformationMode transformationMode)
```

Reply:
Thank you for the feedback. I have never seen this function. I've been scaling thumbnail pixmaps like this in a few places. I can search and replace to switch all of them to this utility function.

Reply:
I will search for an example on how to use your utility function and replace all of the calls throughout the preview and thumbnail systems with it in a follow up PR dedicated to that.